File: 3_javascript/esPalindromo2.js
function esPalindromo2(palabra) {
  let izquierdo = 0;
  let derecho = palabra.length - 1;
  while (izquierdo < derecho) {
    if (palabra[izquierdo].toLowerCase() != palabra[derecho].toLowerCase())
      return false;
    izquierdo++;
    derecho--;
  }
  return true;
}

exports.esPalindromo2 = esPalindromo2;

console.log(esPalindromo2("mex"));
console.log(esPalindromo2("Lol"));

File: 1_Python/es_palindromo2.py
def es_palindromo(palabra): # Complejidad Tiempo O(n) | Espacio O(1)
    izquierdo = 0
    derecho = len(palabra) -1
    while izquierdo < derecho:
        if palabra[izquierdo].lower() != palabra[derecho].lower():
            return False
        izquierdo += 1
        derecho -= 1
    return True

print(es_palindromo("mex"))
print(es_palindromo("Lol"))
